bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,323 @@
+            name="`Total Stars`",
+            value=f"**{kata_information['totalStars']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Completed`",
+            value=f"**{kata_information['totalCompleted']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Attempts`",
+            value=f"**{kata_information['totalAttempts']}**",
+            inline=False,
+        )
+        return miscellaneous_embed
+
+    @staticmethod
+    def create_view(dropdown: InformationDropdown, link: str) -> ui.View:
+        """
+        Creates the discord.py View for the Discord message components (dropdowns and buttons).
+
+        The discord UI is implemented onto the embed, where the user can choose what information about the kata they
+        want, along with a link button to the kata itself.
+        """
+        view = ui.View()
+        view.add_item(dropdown)
+        view.add_item(ui.Button(label="View the Kata", url=link))
+        return view
+
+    @commands.command(aliases=["kata"])
+    async def challenge(self, ctx: commands.Context, language: str = "python", *, query: str = None) -> None:
+        """
+        The challenge command pulls a random kata (challenge) from codewars.com.
+
+        The different ways to use this command are:
+        `.challenge <language>` - Pulls a random challenge within that language's scope.
+        `.challenge <language> <difficulty>` - The difficulty can be from 1-8,
+        1 being the hardest, 8 being the easiest. This pulls a random challenge within that difficulty & language.
+        `.challenge <language> <query>` - Pulls a random challenge with the query provided under the language
+        `.challenge <language> <query>, <difficulty>` - Pulls a random challenge with the query provided,
+        under that difficulty within the language's scope.
+        """
+        if language.lower() not in SUPPORTED_LANGUAGES["stable"] + SUPPORTED_LANGUAGES["beta"]:
+            raise commands.BadArgument("This is not a recognized language on codewars.com!")
+
+        get_kata_link = f"https://codewars.com/kata/search/{language}"
+
+        if language and not query:
+            level = f"-{choice([1, 2, 3, 4, 5, 6, 7, 8])}"
+        elif "," in query:
+            query_splitted = query.split("," if ", " not in query else ", ")
+
+            if len(query_splitted) > 2:
+                raise commands.BadArgument(
+                    "There can only be one comma within the query, separating the difficulty and the query itself."
+                )
+
+            query, level = query_splitted
+            level = f"-{level}"
+        elif query.isnumeric():
+            level, query = f"-{query}", None
+        else:
+            level = None
+
+        params = {}
+        if query:
+            params["q"] = query
+        if level:
+            params["r[]"] = level
```

Comment:
Again, you can even just move this into the if-statements themselves.

So instead of doing `level, query = f"-{query}", None` you can just do `params["q"] = f"-{query}`.